src/monitor/node_active_protocol.c

```diff
@@ -244,7 +246,8 @@ node_active(PG_FUNCTION_ARGS)
 	Oid currentReplicationStateOid = PG_GETARG_OID(5);
 
 	bool currentPgIsRunning = PG_GETARG_BOOL(6);
-	int64 currentWalDelta = PG_GETARG_INT64(7);
+	text *currentLSNText = PG_GETARG_TEXT_P(7);
```

Comment:
I believe we could be `PG_GETARG_LSN` here, given a SQL declaration of the function that takes a `pg_lsn` input. Given the following code in `src/include/utils/pg_lsn.h` in PostgreSQL, we don't seem to have an `LSNOID`, so maybe sending the text representation and using the `INT8OID` to target a function that accept a `pg_lsn` data type would work?

``` C
#define DatumGetLSN(X) ((XLogRecPtr) DatumGetInt64(X))
#define LSNGetDatum(X) (Int64GetDatum((int64) (X)))

#define PG_GETARG_LSN(n)	 DatumGetLSN(PG_GETARG_DATUM(n))
#define PG_RETURN_LSN(x)	 return LSNGetDatum(x)
```